### ISSUE TITLE
Why this error?

### ISSUE THREAD

**Issue by roberto910907** [swiftmailer/swiftmailer#924]
Attempted to call an undefined method named "newInstance" of class "Swift_Message".

And also Im facing problem with the param $localDomain in this commit [5e58f2e7](https://github.com/swiftmailer/swiftmailer/commit/5e58f2e767822167a26d5fdcd6f8a02f526985af)

My symfony version is 2.8 and after composer update:

```
PHP Warning:  Missing argument 4 for Swift_Transport_EsmtpTransport::__construct(), called in /var/www/html/Cortex/releases/20170520005308/app/cache/prod/appProdProjectContainer.php on line 11098 and defined in /var/www/html/Cortex/shared/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/EsmtpTransport.php on line 56
PHP Stack trace:
PHP   1. {main}() /var/www/html/Cortex/releases/20170520005308/app/console:0
PHP   2. Symfony\Component\Console\Application->run() /var/www/html/Cortex/releases/20170520005308/app/console:29
PHP   3. Symfony\Bundle\FrameworkBundle\Console\Application->doRun() /var/www/html/Cortex/shared/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:126
PHP   4. Symfony\Component\HttpKernel\Kernel->boot() /var/www/html/Cortex/shared/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:70
PHP   5. Symfony\Component\HttpKernel\Kernel->initializeContainer() /var/www/html/Cortex/shared/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:133
PHP   6. Symfony\Component\DependencyInjection\Container->get() /var/www/html/Cortex/shared/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:526
PHP   7. appProdProjectContainer->getCacheWarmerService() /var/www/html/Cortex/shared/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:314
PHP   8. Symfony\Component\DependencyInjection\Container->get() /var/www/html/Cortex/releases/20170520005308/app/cache/prod/appProdProjectContainer.php:1410
PHP   9. appProdProjectContainer->getRouterService() /var/www/html/Cortex/shared/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:314
PHP  10. Symfony\Component\DependencyInjection\Container->get() /var/www/html/Cortex/releases/20170520005308/app/cache/prod/appProdProjectContainer.php:9738
PHP  11. appProdProjectContainer->getMonolog_Logger_RouterService() /var/www/html/Cortex/shared/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:314
PHP  12. Symfony\Component\DependencyInjection\Container->get() /var/www/html/Cortex/releases/20170520005308/app/cache/prod/appProdProjectContainer.php:9089
PHP  13. appProdProjectContainer->getMonolog_Handler_MailService() /var/www/html/Cortex/shared/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:314
PHP  14. Symfony\Component\DependencyInjection\Container->get() /var/www/html/Cortex/releases/20170520005308/app/cache/prod/appProdProjectContainer.php:8971
PHP  15. appProdProjectContainer->getMonolog_Handler_BufferedService() /var/www/html/Cortex/shared/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:314
PHP  16. Symfony\Component\DependencyInjection\Container->get() /var/www/html/Cortex/releases/20170520005308/app/cache/prod/appProdProjectContainer.php:8954
PHP  17. appProdProjectContainer->getMonolog_Handler_SwiftService() /var/www/html/Cortex/shared/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:314
PHP  18. Symfony\Component\DependencyInjection\Container->get() /var/www/html/Cortex/releases/20170520005308/app/cache/prod/appProdProjectContainer.php:9004
PHP  19. appProdProjectContainer->getSwiftmailer_Mailer_Default_Transport_RealService() /var/www/html/Cortex/shared/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:314
PHP  20. Swift_Transport_EsmtpTransport->__construct() /var/www/html/Cortex/releases/20170520005308/app/cache/prod/appProdProjectContainer.php:11098
PHP Notice:  Undefined variable: localDomain in /var/www/html/Cortex/shared/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/EsmtpTransport.php on line 58
PHP Stack trace:
PHP   1. {main}() /var/www/html/Cortex/releases/20170520005308/app/console:0
PHP   2. Symfony\Component\Console\Application->run() /var/www/html/Cortex/releases/20170520005308/app/console:29
```

Im just updated my composer and my app its broken now.

**Comment by roberto910907** [comment]
I figure out that in my composer.json "symfony/swiftmailer-bundle": "~2.3", I thought that the **~** operator gets the latest version that is backwards-compatible, but the version 2.6 of swiftmailer-bundle requires swiftmailer 6.0 and egulias/email-validator.

  - Installing egulias/email-validator (2.1.2)
  - Updating swiftmailer/swiftmailer (v5.0.3 => v6.0.0): Downloading (100%) 
  - Removing symfony/swiftmailer-bundle (v2.3.0)         
  - Installing symfony/swiftmailer-bundle (v2.6.0): Downloading (100%) 

I removed the **~** and composer reverts the changes.
Im not a guru of composer, but what was the problem?

**Comment by HeahDude** [comment]
Hello @roberto910907, thank you for the report.

The bundle introduced the support of SwiftMailer 6.0 too early, but don't worry a patch is coming really soon.

In the mean time keeping the bundle to `~2.5.*` is indeed the workaround.

**Comment by fabpot** [comment]
Fixed. I've release 2.6.1 for the bundle which removes support for Swiftmailer 6.0 and released 2.7 with support for it.

**Comment by helmuthva** [comment]
@fabpot:

2.6.1 still pulls 6:

swiftmailer/swiftmailer": ">=4.2.0|~5.0"

best
